ui/app/src/views/SwapPage/useSwapPageData.ts

```diff
@@ -287,6 +287,10 @@ export const useSwapPageData = () => {
         return "Invalid Amount";
       case SwapState.VALID_INPUT:
         return "Swap";
+      case SwapState.FRONTRUN_SLIPPAGE:
```

Comment:
Thanks @shane-moore for the well thought out contribution! As @mccallofthewild said contributions are 100% welcome.

I think the main opportunity for change I see here now is that we have to check against the message within nextStepAllowed to decide validity. It gets brittle.

The only change I'd like is for `nextStepMessage` to return whether the swap is valid.

Perhaps we do some sort of ```const SwapValidityMessage = (isValid: boolean, message: string) => ({ isValid, message })```

Then rename `nextStepMessage` to `nextStepValidityMessage` and return SwapValidityMessage objects

Then nextStepMessage just returns a computed that gets `nextStepValidityMessage.value.message` and nextStepAllowed returns `nextStepValidityMessage.value.isValid`

Reply:
@mccallofthewild cool 🤠 

My Keplr wallet address is sif1wfmnamflj4yrtvm9n8edm2dact5lz9pmzssrce, in case air drop is possible. Otherwise, just let me know how to to proceed in order to gain the full power of testnet.

@ajoslin, solid call as to add the isValid flag and check against that to determine whether the user can proceed. Then, the message property can independently handle what text to display.  This was implemented in my latest commit, which seems to encompass your suggestions wholly, but let me know if more tweaks would lead to increased robustness.